crates/bevy_asset/src/loader_builders.rs

```diff
@@ -74,47 +210,108 @@ impl<'ctx, 'builder> NestedLoader<'ctx, 'builder> {
+        NestedLoader {
+            load_context: self.load_context,
+            meta_transform: self.meta_transform,
+            typing: UnknownType { _priv: () },
+            mode: self.mode,
         }
     }
 
-    /// Load assets without static type information.
+    // convert between `M`s
+
+    /// When [`load`]ing, create only asset handles, rather than loading the
+    /// actual asset.
     ///
-    /// If you need to specify the type of asset, but cannot do it statically,
-    /// use `.with_asset_type_id()`.
+    /// [`load`]: Self::load
+    pub fn deferred(self) -> NestedLoader<'ctx, 'builder, T, Deferred> {
+        NestedLoader {
+            load_context: self.load_context,
+            meta_transform: self.meta_transform,
+            typing: self.typing,
+            mode: Deferred { _priv: () },
+        }
+    }
+    /// The [`load`] call itself will load an asset, rather than scheduling the
+    /// loading to happen later.
+    ///
+    /// This gives you access to the loaded asset, but requires you to be in an
+    /// async context, and be able to `await` the resulting future.
+    ///
+    /// [`load`]: Self::load
     #[must_use]
-    pub fn untyped(self) -> UntypedNestedLoader<'ctx, 'builder> {
-        UntypedNestedLoader { base: self }
+    pub fn immediate<'c>(self) -> NestedLoader<'ctx, 'builder, T, Immediate<'builder, 'c>> {
+        NestedLoader {
+            load_context: self.load_context,
+            meta_transform: self.meta_transform,
+            typing: self.typing,
+            mode: Immediate { reader: None },
+        }
     }
+}
+
+// deferred loading logic
 
-    /// Retrieves a handle for the asset at the given path and adds that path as a dependency of the asset.
-    /// If the current context is a normal [`AssetServer::load`](crate::AssetServer::load), an actual asset
-    /// load will be kicked off immediately, which ensures the load happens as soon as possible.
-    /// "Normal loads" kicked from within a normal Bevy App will generally configure the context to kick off
-    /// loads immediately.
-    /// If the current context is configured to not load dependencies automatically
-    /// (ex: [`AssetProcessor`](crate::processor::AssetProcessor)),
-    /// a load will not be kicked off automatically. It is then the calling context's responsibility to begin
-    /// a load if necessary.
+impl NestedLoader<'_, '_, Typed, Deferred> {
+    /// Retrieves a handle for the asset at the given path and adds that path as
+    /// a dependency of this asset.
+    ///
+    /// This requires you to know the type of asset statically.
+    /// - If you have runtime info for what type of asset you're loading (e.g. a
+    ///   [`TypeId`]), use [`erased`].
+    /// - If you do not know at all what type of asset you're loading, use
+    ///   [`unknown_type`].
+    ///
+    /// [`erased`]: Self::erased
+    /// [`unknown_type`]: Self::unknown_type
```

Comment:
This line doesn't match, so if you don't rename `of_unknown_type`, fixup this doc.

Reply:
Left it as `unknown_type`